Code/Editor/CryEdit.cpp

```diff
@@ -2274,19 +2274,20 @@ int CCryEditApp::IdleProcessing(bool bBackgroundUpdate)
 
             GetIEditor()->Notify(eNotify_OnIdleUpdate);
         }
-
-        AZ::ComponentApplication* componentApplication = nullptr;
-        AZ::ComponentApplicationBus::BroadcastResult(componentApplication, &AZ::ComponentApplicationRequests::GetApplication);
-        if (componentApplication)
-        {
-            componentApplication->TickSystem();
-        }
     }
     else if (GetIEditor()->GetSystem() && GetIEditor()->GetSystem()->GetILog())
     {
         GetIEditor()->GetSystem()->GetILog()->Update(); // print messages from other threads
     }
 
+    // Tick System Events, even in the background
+    AZ::ComponentApplication* componentApplication = nullptr;
+    AZ::ComponentApplicationBus::BroadcastResult(componentApplication, &AZ::ComponentApplicationRequests::GetApplication);
+    if (componentApplication)
+    {
+        componentApplication->TickSystem();
+    }
```

Comment:
You can asset the ComponentApplicationRequest via the `AZ::Interface<AZ::ComponentApplicationRequests>`
```suggestion
    AZ::ComponentApplicationRequest* componentApplication = AZ::Interface<AZ::ComponentApplicationRequests>::Get();
    if (componentApplication)
    {
        componentApplication->TickSystem();
    }
```

Reply:
Ah nice! I'll switch to the AZ::Interface approach then.